homeassistant/components/soundtouch/media_player.py

```diff
@@ -232,17 +155,15 @@ def volume_level(self):
         """Volume level of the media player (0..1)."""
         return self._volume.actual / 100
 
-    @property
-    def name(self):
-        """Return the name of the device."""
-        return self._name
-
     @property
     def state(self):
         """Return the state of the device."""
-        if self._status.source == "STANDBY":
+        if self._status is None or self._status.source == "STANDBY":
             return STATE_OFF
 
+        if self._status.source == "INVALID_SOURCE":
+            return STATE_UNKNOWN
```

Comment:
We should return `None` as state to represent unknown state. It will be converted by the base entity class.